source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -670,6 +730,43 @@ logical sessions are enabled in the deployment. Clients MUST check for this
 field and set the ServerDescription's logicalSessionTimeoutMinutes field to this
 value, or to null otherwise.
 
+topologyVersion
+```````````````
+
+MongoDB 4.4 and later include a ``topologyVersion`` field in all isMaster
+and State Change Error responses. Clients MUST check for this field in both
+"ok: 1" and "ok: 0" replies and set the ServerDescription's topologyVersion
+field to this value, or to null otherwise. The topologyVersion helps the
+client and server determine the relative freshness of topology information
+in concurrent messages. (See `What is the purpose of topologyVersion?`_)
+
+The topologyVersion is a subdocument with two fields, "processId" and
+"counter":
+
+.. code:: typescript
+
+    {
+        topologyVersion: {processId: <ObjectId>, counter: <int64>},
+        ( ... other fields ...)
+    }
+
+topologyVersion Comparison
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To compare a topologyVersion from an isMaster or State Change Error
```

Comment:
I know we discussed this in the google doc...

Comparing this algorithm with the one in the `topologyVersion Comparison` section above, they seem identical, except this one defines how to compare two non-null topology versions in more detail.  Can we DRY this up?  As an implementor I'm not clear how to proceed and where I would use the preceding definition of the comparison logic and where I'd use this one.